### PR TITLE
Tests For Debug Page PR #852

### LINKED ISSUE #738: [Ember-Migration] Create Debug page
### Requirements
As we are migrating towards Emberjs we need the given section to be sync with our production, to look it exactly like production version. In this ticket we have to make the mentioned section UI as well as functionalities to match same as in production version.

#### In vanilla site (currently running www site in production, created with plain HTML, CSS, JS)
Debug page -  https://www.realdevsquad.com/debug

#### In beta site(created with EmberJS)
Doesn't exists

### Important links

- https://staging-www.realdevsquad.com/ (This one is made with Ember js, staging one)
- https://www.realdevsquad.com/ (This one is the production site)

### Additional Note

- Design should be same as in prod.
- Take reference from the prod site wherever needed.
- It should be responsive.
- Code quality should be maintained.
- Use `ember generate` commands to generate `components`, `templates` and other files.

### Questions you can have in mind

1. Is this should be under feature flag?
Yes, it should be under feature dev=true

2. What is feature flag?
It's a simple variable which we can use to turn off/on the feature. Currently we use dev=true in URL query.

3. Do you have to use the same code as in production site?
No, the solution is upto you how you want to approach the problem. It should be same UI wise and functionality wise, rest other implementation is on you.

4. In which branch ember code is in?
It's in `develop-ember` branch, make sure to checkout to that branch.

### LINKED ISSUE #854: Tests For Debug Page
### Issue Description

Add tests for debug page, PR: #852

### PR DESCRIPTION
**Pull request by ajoykumardas12**
Date: `02/02/2024`

Developer Name: @ajoykumardas12 

---

## Issue Ticket Number:-

- closes #854 
- closes #738 

## Description:

This PR contains tests for #852 
This PR also contains changes from #852 which are reviewed there already.

Haven't included test for util function `isoToLocalDate` which is timezone dependent so would be different results for local, Github CI or even on local with different timezone.

Is Under Feature Flag

- [x] Yes
- [ ] No

Database changes

- [ ] Yes
- [x] No

Breaking changes (If your feature is breaking/missing something please mention pending tickets)

- [ ] Yes
- [x] No

Is Development Tested?

- [x] Yes
- [ ] No

Screenshot_tests

### Add relevant Screenshot below ( e.g test coverage etc. )

NA
For visuals check #852

### CODE REVIEW

File: app/components/debug-grids.hbs
Comment (ajoykumardas12) [inline app/components/debug-grids.hbs:1]: The new changes in this file after #852 are `data-test-` attributes

File: app/templates/debug.hbs
Comment (ajoykumardas12) [inline app/templates/debug.hbs:1]: Added data-test attribute

File: tests/constants/users-data.js
Comment (ajoykumardas12) [inline tests/constants/users-data.js:1]: New file.

File: tests/integration/components/debug-grids-test.js
Comment (ajoykumardas12) [inline tests/integration/components/debug-grids-test.js:1]: Changed - needs review

File: tests/unit/routes/debug-test.js
Comment (ajoykumardas12) [inline tests/unit/routes/debug-test.js:1]: Changed - needs review.

File: tests/unit/utils/iso-to-local-date-test.js
Comment (ajoykumardas12) [inline tests/unit/utils/iso-to-local-date-test.js:1]: New test - needs review.

File: tests/integration/components/debug-grids-test.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        'src',
+        'https://res.cloudinary.com/realdevsquad/image/upload/mock-image.png',
+      );
+    assert.dom('[data-test-debug-name]').hasText('John Doe');
+    assert
+      .dom('[data-test-debug-social-id="Twitter Id"]')
+      .hasText('Twitter Id: N/A');
+    assert
+      .dom('[data-test-debug-social-id="Linkedin Id"]')
+      .hasText('Linkedin Id: https://www.linkedin.com/in/john-doe/');
```
[the remark this answers is not in the document]
Comment (satyam73) [reply tests/integration/components/debug-grids-test.js:47]: Similarly we can do in other places too

File: tests/integration/components/debug-grids-test.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  test('it shows error message when authentication fails', async function (assert) {
+    set(this.owner.lookup('service:store'), 'findRecord', () => {
+      return;
+    });
+
+    await render(hbs`<DebugGrids />`);
+
+    assert.dom('[data-test-debug-grids]').doesNotExist();
+    assert.dom('[data-test-debug-error-message]').exists();
+  });
```
[the remark this answers is not in the document]
Comment (ajoykumardas12) [reply tests/integration/components/debug-grids-test.js]: It shows error message associated with this test: `it shows error message when authentication fails`

File: tests/integration/components/debug-grids-test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ajoykumardas12) [reply tests/integration/components/debug-grids-test.js]: Thanks, changed.

File: app/components/debug-grids.hbs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            </label>
+          </div>
+        </div>
+      {{/if}}
+    </div>
+  </div>
+{{else}}
+  <div data-test-debug-error-message class="debug__error-message">
+    <p>Something went wrong... </p>
+    <p>Please try again.</p>
```
Comment (satyam73) [inline app/components/debug-grids.hbs:99]: Error should be on point and guide user what to do next(if possible there should be navigation to the home)

Comment (ajoykumardas12) [reply app/components/debug-grids.hbs:99]: `Please Sign In to load debug data` is this message fine?

**Review by satyam73 — DISMISSED** [review]
LGTM

**Review by Achintya-Chatterjee — APPROVED** [review]

**Comment by ajoykumardas12** [comment]
Added fix for a bug from the main PR #852  found on staging

**Review by Achintya-Chatterjee — APPROVED** [review]

**Review by satyam73 — APPROVED** [review]

**Pull request merged**